tests/frame/to_native_test.py

```diff
@@ -0,0 +1,11 @@
+from typing import Any
+
+import narwhals.stable.v1 as nw
+
+
+def test_to_native(constructor_eager: Any) -> None:
+    data = {"a": [1, 3, 2], "b": [4, 4, 6], "z": [7.1, 8, 9]}
+    df_raw = constructor_eager(data)
+    df = nw.from_native(df_raw, eager_only=True)
+
+    assert isinstance(df.to_native(), df_raw.__class__)
```

Comment:
I do believe that this does not exactly check what you need it to check because if the type of `df.tonative()` is actually a subclass of `df_raw.__class__`, it will also pass. While usually `isinstance` is preferred over `type()` comparison, I think it does warrant implementation here: `assert type(df.to_native()) == df_raw.__class__`, where you would need to put `# noqa: E721`.

PS: also see https://stackoverflow.com/a/9611031/10367342

Reply:
true, but none of the dataframes we support are subclasses of each other